### PR TITLE
Adds CBMC proofs for aws_cryptosdk_priv_hdr_parse_{frame,iv}_len 

### PR DESCRIPTION
**Pull request by tegansb**
Adds harness, Makefile and yaml for aws_cryptosdk_priv_hdr_parse_frame_len and aws_cryptosdk_priv_hdr_parse_iv_len.

Proofs are for split version of hdr_parse in #641.
Uses ensures, is bounded and validity functions from #632.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

# Check any applicable:
- [ ] Were any files moved? Moving files changes their URL, which breaks all hyperlinks to the files.

### CODE REVIEW

**Review by alex-chew — APPROVED** [review]
Overall LGTM. I think we can [pull some assumptions/preconditions out into a helper function](https://github.com/aws/aws-encryption-sdk-c/pull/644#discussion_r538953400), but it's not a blocker for this PR.

File: verification/cbmc/proofs/aws_cryptosdk_priv_hdr_parse_frame_len/Makefile [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+REMOVE_FUNCTION_BODIES += --remove-function-body nondet_compare
+
+# Functions removed for enc_ctx_clear
+REMOVE_FUNCTION_BODIES += aws_hash_callback_string_destroy
+REMOVE_FUNCTION_BODIES += aws_hash_string
+REMOVE_FUNCTION_BODIES += aws_raise_error_private
+REMOVE_FUNCTION_BODIES += aws_string_bytes
+REMOVE_FUNCTION_BODIES += aws_string_destroy
+REMOVE_FUNCTION_BODIES += aws_string_is_valid
+REMOVE_FUNCTION_BODIES += hashlittle2
```
Comment (adpaco-aws) [inline verification/cbmc/proofs/aws_cryptosdk_priv_hdr_parse_frame_len/Makefile:73]: The correct variable is `REMOVE_FUNCTION_BODY`. With the starter kit, you do not need to specify the flag `--remove-function-body` anymore.

File: verification/cbmc/proofs/aws_cryptosdk_priv_hdr_parse_iv_len/Makefile [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+REMOVE_FUNCTION_BODIES += --remove-function-body nondet_compare
+
+# Functions removed for enc_ctx_clear
+REMOVE_FUNCTION_BODIES += aws_hash_callback_string_destroy
+REMOVE_FUNCTION_BODIES += aws_hash_string
+REMOVE_FUNCTION_BODIES += aws_raise_error_private
+REMOVE_FUNCTION_BODIES += aws_string_bytes
+REMOVE_FUNCTION_BODIES += aws_string_destroy
+REMOVE_FUNCTION_BODIES += aws_string_is_valid
+REMOVE_FUNCTION_BODIES += hashlittle2
```
Comment (adpaco-aws) [inline verification/cbmc/proofs/aws_cryptosdk_priv_hdr_parse_iv_len/Makefile:73]: Same as above.

**Review by adpaco-aws — CHANGES_REQUESTED** [review]
The proof is trying to remove function bodies from a handful of functions but this should not be working. Please correct this and check if the result is expected.

File: verification/cbmc/proofs/aws_cryptosdk_priv_hdr_parse_frame_len/Makefile (hunk 1 above)
Comment (tegansb) [reply verification/cbmc/proofs/aws_cryptosdk_priv_hdr_parse_frame_len/Makefile]: Updated here and in other functions as well.

File: verification/cbmc/proofs/aws_cryptosdk_priv_hdr_parse_iv_len/Makefile (hunk 2 above)
Comment (tegansb) [reply verification/cbmc/proofs/aws_cryptosdk_priv_hdr_parse_iv_len/Makefile]: Updated!

**Review by feliperodri — APPROVED** [review]
LGTM, but you need to update this branch. @tegansb

**Review by adpaco-aws — APPROVED** [review]

**Pull request merged**